### PR TITLE
select extra options in quickbench

### PR DESCRIPTION
**Pull request by partouf**
tries to include;
* gnu/clang
* c++ version
* c++lib

### CODE REVIEW

File: static/panes/editor.js [hunk 1]
```diff
@@ -500,6 +519,55 @@ Editor.prototype.updateOpenInQuickBench = function () {
         text: this.getSource()
     };
 
+    var compilers = this.getCompilerStates();
+
+    _.each(compilers, function (compiler) {
+        var knownCompiler = false;
+        if (compiler.compiler.match(/^g[0-9]*$/)) {
+            knownCompiler = true;
+            quickBenchState.compiler = "gcc-" + compiler.compiler[1] + "." + compiler.compiler[2];
```
Comment (apmorton) [inline static/panes/editor.js]: these will break for gcc/clang 10

maybe we can add something in the server side properties?
```
compiler.g92.quickBenchCompiler=gcc-9.2
```

we could then set defaults for older gcc/clang versions:
```
group.gcc86.quickBenchCompiler=gcc-5.5
```

File: static/panes/editor.js [hunk 2]
```diff
@@ -500,6 +519,55 @@ Editor.prototype.updateOpenInQuickBench = function () {
...
+        } else if (compiler.compiler.match(/^clang[0-9]*x?$/)) {
+            knownCompiler = true;
+            quickBenchState.compiler = "clang-" + compiler.compiler[5] + "." + compiler.compiler[6];
+        }
+
+        if (knownCompiler) {
+            if (compiler.options.indexOf("-O3") !== -1) {
```
Comment (apmorton) [inline static/panes/editor.js]:
```javascript
var match = compiler.options.match(/-(O([0-3sg]|fast))/);
if (match !== null)
  quickBenchState.optim = match[1];
```

regex not tested, but something like that could collapse these if chains down to a single statement
same for the cppVersion stuff below, although you would have to have to handle the edge cases for `2a`

File: static/panes/editor.js (hunk 1 above)
Comment (partouf) [reply static/panes/editor.js]: hmm, didn't think of that.

Comment (mattgodbolt) [reply static/panes/editor.js]: Thanks both: good catch @apmorton but great start @partouf

File: static/panes/editor.js [hunk 3]
```diff
@@ -500,6 +519,44 @@ Editor.prototype.updateOpenInQuickBench = function () {
...
+            quickBenchState.compiler = "clang-" + compiler.compiler[5] + "." + compiler.compiler[6];
+        }
+
+        if (knownCompiler) {
+            var match = compiler.options.match(/-(O([0-3sg]|fast))/);
+            if (match !== null) {
+                quickBenchState.optim = match[1];
+            }
+
+            if ((compiler.options.indexOf("-std=c++11") !== -1) ||
```
Comment (mattgodbolt) [inline static/panes/editor.js]: Nice :) thanks!

File: static/panes/editor.js (hunk 1 above)
Comment (partouf) [reply static/panes/editor.js]: Changed to use semver instead, felt uncomfortable with adding a special property for it. Though we can always still do that if this gives us too much trouble.

**Comment by mattgodbolt** [comment]
Let us know if you need another look @partouf else I'd love to get this merged :)

**Comment by apmorton** [comment]
just tried this locally and the optimization level doesn't appear to work for me - the data appears in the base64 json, but the level isn't selected on the quickbench site, so I guess the issue is with the code added on their end, not ours.

**Comment by partouf** [comment]
It seems that the options for optimization are:
* G
* F
* S
* 0
* 1
* 2
* 3

So 0-3 should have worked, maybe? Though the GFS wouldn't have worked.
I am noticing they explicitly need to be strings.

**Comment by partouf** [comment]
Fixed the optimization level issue

**Comment by mattgodbolt** [comment]
Hey @partouf :) Let us know when you're ready to merge, this looks good to me!

**Pull request merged**